src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -398,10 +398,11 @@ function updateDOMProperties(
     } else if (propKey === DANGEROUSLY_SET_INNER_HTML) {
       var nextHtml = nextProp ? nextProp[HTML] : undefined;
       var lastHtml = lastProp ? lastProp[HTML] : undefined;
-      if (nextHtml) {
+      if (nextHtml || nextHtml === 0 || nextHtml === '') {
```

Comment:
Maybe do a typeof instead? that is what we have done in other places where we need to detect numbers / strings. I am not sure this works with negative values